Mathlib/Data/Fin/Basic.lean

```diff
@@ -2352,130 +2397,117 @@ theorem succ_above_pred_above {p : Fin n} {i : Fin (n + 1)} (h : i ≠ p.cast_su
         rw [← Nat.succ_lt_succ_iff] at h₀
-        simpa only [coe_succ] using h₀
-      simp only [coe_cast_pred b h₀, coe_cast_pred b.succ h₁, coe_succ]
-#align fin.succ_pred_above_succ Fin.succ_pred_above_succ
+        simpa only [val_succ] using h₀
+      simp only [coe_castPred b h₀, coe_castPred b.succ h₁, val_succ]
+#align fin.succ_pred_above_succ Fin.succ_predAbove_succ
 
 @[simp]
-theorem cast_pred_cast_succ (i : Fin (n + 1)) : castPred i.cast_succ = i := by
-  simp [cast_pred, pred_above, le_last]
-#align fin.cast_pred_cast_succ Fin.cast_pred_cast_succ
+theorem castPred_castSucc (i : Fin (n + 1)) : castPred (castSucc i) = i := by
+  simp [castPred, predAbove, not_lt.mpr (le_last i)]
+#align fin.cast_pred_cast_succ Fin.castPred_castSucc
 
-theorem cast_succ_cast_pred {i : Fin (n + 2)} (h : i < last _) : castSucc i.cast_pred = i :=
-  by
-  rw [cast_pred, pred_above, dif_neg]
+theorem castSucc_castPred {i : Fin (n + 2)} (h : i < last (n + 1)) : castSucc i.castPred = i := by
+  rw [castPred, predAbove, dif_neg]
   · simp [Fin.eq_iff_veq]
   · exact h.not_le
-#align fin.cast_succ_cast_pred Fin.cast_succ_cast_pred
+#align fin.cast_succ_cast_pred Fin.castSucc_castPred
 
-theorem coe_cast_pred_le_self (i : Fin (n + 2)) : (i.cast_pred : ℕ) ≤ i :=
-  by
+theorem coe_castPred_le_self (i : Fin (n + 2)) : (i.castPred : ℕ) ≤ i := by
   rcases i.le_last.eq_or_lt with (rfl | h)
   · simp
-  · rw [cast_pred, pred_above, dif_neg]
+  · rw [castPred, predAbove, dif_neg]
     · simp
-    · simpa [lt_iff_coe_lt_coe, le_iff_coe_le_coe, lt_succ_iff] using h
-#align fin.coe_cast_pred_le_self Fin.coe_cast_pred_le_self
+    · simpa [lt_iff_val_lt_val, le_iff_val_le_val, lt_succ_iff] using h
+#align fin.coe_cast_pred_le_self Fin.coe_castPred_le_self
 
-theorem coe_cast_pred_lt_iff {i : Fin (n + 2)} : (i.cast_pred : ℕ) < i ↔ i = Fin.last _ :=
-  by
+theorem coe_castPred_lt_iff {i : Fin (n + 2)} : (i.castPred : ℕ) < i ↔ i = Fin.last _ := by
   rcases i.le_last.eq_or_lt with (rfl | H)
   · simp
-  · simp only [ne_of_lt H]
-    rw [← cast_succ_cast_pred H]
+  · simp only [_root_.ne_of_lt H]
+    rw [← castSucc_castPred H]
     simp
-#align fin.coe_cast_pred_lt_iff Fin.coe_cast_pred_lt_iff
+#align fin.coe_cast_pred_lt_iff Fin.coe_castPred_lt_iff
 
-theorem lt_last_iff_coe_cast_pred {i : Fin (n + 2)} : i < Fin.last _ ↔ (i.cast_pred : ℕ) = i :=
-  by
+theorem lt_last_iff_coe_castPred {i : Fin (n + 2)} :
+    i < Fin.last _ ↔ (i.castPred : ℕ) = i := by
   rcases i.le_last.eq_or_lt with (rfl | H)
   · simp
   · simp only [H]
-    rw [← cast_succ_cast_pred H]
+    rw [← castSucc_castPred H]
     simp
-#align fin.lt_last_iff_coe_cast_pred Fin.lt_last_iff_coe_cast_pred
+#align fin.lt_last_iff_coe_cast_pred Fin.lt_last_iff_coe_castPred
 
 end PredAbove
 
-/-- `min n m` as an element of `fin (m + 1)` -/
+/-- `min n m` as an element of `Fin (m + 1)` -/
 def clamp (n m : ℕ) : Fin (m + 1) :=
-  of_nat <| min n m
+  OfNat.ofNat <| min n m
```

Comment:
@digama0, are direct invocations of `ofNat` without a literal a bad idea?

Reply:
I would say yes, you should use a different function like `Nat.cast` for non-literals. Using `ofNat` will make some tactics get confused about the term.

Reply:
fixed in #1521